admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
-                  <a href="<?php echo zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'set_display_categories_dropdown=0' . (isset($_GET['cID']) ? '&cID=' . (int)$_GET['cID'] : '') . '&cPath=' . $cPath . (isset($_GET['page']) ? '&page=' . $_GET['page'] : '')); ?>" title="<?php echo IMAGE_ICON_STATUS_ON; ?>"><i class="fa fa-check fa-lg txt-status-on"></i></a>
-                  <strong><?php echo HEADING_TITLE_GOTO; ?></strong>
+                <div class="col-md-4">
+                    <?php
+                    echo zen_draw_form('searchForm', FILENAME_CATEGORY_PRODUCT_LISTING, '', 'get', 'class="form-horizontal"'); ?>
+                    <?php
+                    echo zen_hide_session_id(); ?>
+                    <div class="form-group">
+                        <?php
+                        echo zen_draw_label(HEADING_TITLE_SEARCH_DETAIL, 'search', 'class="col-sm-6 col-md-4 control-label"'); ?>
+                        <div class="col-sm-6 col-md-8">
+                            <?php
+                            echo zen_draw_input_field('search', '', 'autofocus="autofocus" class="form-control" id="search"'); ?>
+                        </div>
+                    </div>
+                    <?php
+                    if ($search_result) {
+                        ?>
+                        <div class="form-group">
+                            <div class="col-sm-6 col-md-4 control-label"><?php
+                                echo TEXT_INFO_SEARCH_DETAIL_FILTER; ?></div>
+                            <div class="col-sm-6 col-md-8">
+                                <strong>"<?php
+                                    echo zen_output_string_protected($_GET['search']); ?>"</strong>
+                                <a href="<?php
+                                echo zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING); ?>" class="btn btn-default" role="button"><?php
+                                    echo IMAGE_RESET; ?></a>
+                            </div>
+                        </div>
+                    <?php
+                    } ?>
+                    <?php
+                    echo '</form>'; ?>
+                    <?php
+                    echo zen_draw_form('goto', FILENAME_CATEGORY_PRODUCT_LISTING, '', 'get', 'class="form-horizontal"'); ?>
+                    <?php
+                    echo zen_hide_session_id(); ?>
+                    <div class="form-group">
+                        <?php
+                        if ($_SESSION['display_categories_dropdown'] === 0) { ?>
+                            <div class="col-sm-6 col-md-4 control-label">
+                                <a href="<?php
+                                echo zen_href_link(
+                                    FILENAME_CATEGORY_PRODUCT_LISTING,
+                                    'set_display_categories_dropdown=1' . (isset($_GET['cID']) ? '&cID=' . (int)$_GET['cID'] : '') . '&cPath=' . $cPath . (isset($_GET['page']) ? '&page='
```

Comment:
$_GET['cID'] is already sanitized to an int.

Reply:
ok, next PR